language/vm/src/views.rs

```diff
@@ -509,6 +509,58 @@ impl<'a, T: ModuleAccess> LocalsSignatureView<'a, T> {
     }
 }
 
+/// Returns the kind of the signature token in the given context (module, function/struct).
+/// The context is needed to determine the kinds of structs & type variables.
+// TODO: rework views to make the interfaces here cleaner.
+pub fn kind_impl(
```

Comment:
Less boilerplate if we put this code directly in `impl SignatureToken` instead?

Reply:
I assume you meant `impl SignatureTokenView`. This is less favorable since `SignatureTokenView` is parametric over a lifetime. If we put it in the impl we would have to specify the lifetime at call site:
```
SignatureTokenView::kind_impl::<'a>(...)
```

Reply:
I meant in `impl SignatureToken`, which does not have a generic type parameter.

Reply:
Ah sorry I confused myself. Sounds good to me.